basic-app/components/wallet/RegisterWallet.tsx

```diff
@@ -16,28 +17,46 @@ gql`
     }
   }
 `
+interface Props {
+  blockchain: Blockchain
+}
 
-export function RegisterWallet() {
+export function RegisterWallet({ blockchain }: Props) {
   const flowUser = useFlowUser()
+  const metamask = useMetaMask()
 
   const { mutate: registerWallet, error, isLoading } = useWalletMutation(useRegisterWalletMutation)
 
   // When the user logs in, register their wallet
   useEffect(() => {
-    if (!flowUser?.addr) {
+    if (blockchain !== Blockchain.Flow || !flowUser?.addr) {
       return
     }
 
     registerWallet({ address: flowUser.addr })
-  }, [flowUser?.addr, flowUser?.loggedIn, registerWallet])
+  }, [blockchain, flowUser?.addr, flowUser?.loggedIn, registerWallet])
+
+  const handleRegister = async () => {
+    switch (blockchain) {
+      case Blockchain.Flow: {
+        fcl.logIn()
+      }
+      case Blockchain.Polygon: {
+        const wallets = await metamask.connect()
+        if (wallets?.length > 0) {
+          registerWallet({ address: wallets[0] })
+        }
```

Comment:
Shouldn't this be in a `useEffect`?

Reply:
The flow one was because we dont have promise. This only happens once for the event so its fine here

Reply:
Can you add a comment in the useEffect as well as here for that?